__tests__/app.test.js

```diff
@@ -0,0 +1,40 @@
+const app = require('../app');
+const runSeed = require('../db/seeds/seed');
+const testData = require('../db/data/test-data');
+const request = require('supertest');
+beforeEach(() => runSeed(testData));
+describe('GET non-existent endpoint', () => {
+    test("returns 404", () => {
+        return request(app)
+        .get('/api/cagetories')
+        .expect(404);
+    });
+    test("returns error message", () => {
+        return request(app)
+        .get('/api/test')
+        .then((res) => {
+            expect(res.body).toEqual({ msg: "error: invalid path" });
+        })
+    })
+})
+describe('GET /api/categories', () => {
+    test("returns status code 200", () => {
+        return request(app)
+        .get('/api/categories')
+        .expect(200);
+    });
+    test("returns array of objects", () => {
+        return request(app)
+        .get('/api/categories')
+        .then((res) => {
+            expect(Array.isArray(res.body.result)).toBe(true);
+        })
+    })
+    test("returns correct response", () => {
+        return request(app)
+        .get('/api/categories')
+        .then((res) => {
+            expect(res.body.result).toMatchObject(testData.categoryData);
```

Comment:
Hey Ibrahim, for any future tests I want you to explicitly check the shape of the data coming back, so checking that properties are there not just checking it matches the unseeded data.

Reply:
In future please add an assertion checking the length of the returned array too when we're GETting multiple resources